src/fu-usb-device.c

```diff
@@ -170,6 +170,7 @@ fu_usb_device_open (FuDevice *device, GError **error)
 	if (idx != 0x00) {
 		g_autofree gchar *tmp = NULL;
 		tmp = g_usb_device_get_string_descriptor (priv->usb_device, idx, NULL);
+		fu_device_set_version_format (device, fu_common_version_guess_format (tmp));
```

Comment:
Thinking a bit more about this, I think we should do `fu_device_set_version_format (device, FU_VERSION_FORMAT_UNKNOWN);` instead and leave all the version guessing logic in `fu_device_set_version()`. With that change, LGTM.